typescript/web/src/components/labeling-tool/options-tool-bar/__tests__/edit-label-class-menu.tsx

```diff
@@ -1,167 +1,81 @@
-  datasetId: string = testDatasetId
-) => {
-  // @ts-ignore
-  return client.mutateOriginal({
-    mutation: CREATE_TEST_DATASET_MUTATION,
-    variables: {
-      name,
-      datasetId,
-      workspaceSlug: "local",
-    },
-    fetchPolicy: "no-cache",
-  });
-};
-
-const renderEditLabelClassMenu = () => {
+const renderEditLabelClassMenu = (mockLink: WildcardMockLink) => {
   return render(<EditLabelClassMenu />, {
-    wrapper: ({ children }) => (
-      <ApolloProvider client={client}>{children}</ApolloProvider>
-    ),
+    wrapper: getApolloMockWrapper(mockLink),
   });
 };
 
-beforeEach(async () => {
-  useLabelingStore.setState({
-    selectedLabelId: "my label id",
-    selectedTool: Tools.SELECTION,
-  });
-
-  await createDataset("Test dataset");
-
-  // @ts-ignore
-  await client.mutateOriginal({
-    mutation: CREATE_LABEL_CLASS_MUTATION,
-    variables: {
-      id: "existing label class id",
-      name: "existing label class",
-      color: "0xaa45f7",
-      datasetId: testDatasetId,
-    },
-  });
-
+beforeEach(() => {
+  act(() =>
+    useLabelingStore.setState({
+      selectedLabelId: BASIC_LABEL_DATA.id,
+      selectedTool: Tools.SELECTION,
+    })
+  );
   jest.clearAllMocks();
 });
 
 it("should create a class", async () => {
-  renderEditLabelClassMenu();
-
-  userEvent.type(screen.getByPlaceholderText(/Search/), "newClass{enter}");
-
-  await waitFor(() => {
-    expect(client.mutate).toHaveBeenCalledWith(
-      expect.objectContaining({
-        variables: {
-          data: expect.objectContaining({
-            name: "newClass",
-          }),
-        },
-      })
-    );
-  });
-
-  // The id is generated by the effect. We retrieve it from the first mutation
-  // which is "createLabelClass"
-  const { id } = (client.mutate as jest.Mock).mock.calls[0][0].variables.data;
-
-  await waitFor(() => {
-    expect(client.mutate).toHaveBeenCalledWith(
-      expect.objectContaining({
-        variables: {
-          data: {
-            labelClassId: id,
-          },
-          where: { id: "my label id" },
-        },
-      })
-    );
-  });
+  const mockLink = getApolloMockLink(APOLLO_MOCKS);
+  renderEditLabelClassMenu(mockLink);
+  act(() =>
+    userEvent.type(screen.getByPlaceholderText(/Search/), "newClassTest")
+  );
+  await waitFor(() => expect(screen.getByText("Create class")).toBeDefined());
+  userEvent.click(screen.getByText("Create class"));
+  await act(() => mockLink.waitForAllResponsesRecursively());
+  expect(CREATE_LABEL_CLASS_ACTION_MOCK.result).toHaveBeenCalled();
+  expect(UPDATE_LABEL_CLASS_ACTION_MOCK.result).toHaveBeenCalled();
 });
 
 it("should change a class", async () => {
-  renderEditLabelClassMenu();
-
+  const mockLink = getApolloMockLink(APOLLO_MOCKS);
+  renderEditLabelClassMenu(mockLink);
   await waitFor(() =>
-    expect(screen.getByText(/existing label class/)).toBeDefined()
+    expect(
+      screen.getByText(DEEP_DATASET_WITH_CLASSES_DATA.labelClasses[1].name)
```

Comment:
You should move `DEEP_DATASET_WITH_CLASSES_DATA.labelClasses[1]` in a variable or directly use the correct LabelClass fixture if it exists (otherwise extract it).

Reply:
This should be fixed with the refactor of `data.fixtures.ts` that you suggested in your other comment, in order to remove relations/nested data.